libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccount.java

```diff
@@ -452,7 +452,7 @@ public String getUserLevelStoragePath() {
 	 */
 	public String getCommunityLevelStoragePath() {
 		String leafDir = INTERNAL_COMMUNITY_PATH;
-		if (!TextUtils.isEmpty(communityId) && !communityId.equals(INTERNAL_COMMUNITY_ID)) {
+		if (!TextUtils.isEmpty(communityId) && !communityId.startsWith(INTERNAL_COMMUNITY_ID)) {
```

Comment:
It's meant to be interoperable on the server because that's internal code. 15-char ID is an internal only representation of IDs. External facing entity IDs are required to be 18-char only, so IDs coming back from a public API endpoint, such as the token endpoint should be 18-char only.